install.sh

```diff
@@ -0,0 +1,142 @@
+                INSTALL_SPECIFIC_VERSION="1"
+                shift
+                ;;
+            *)
+               echo "$1 is not a recognized flag!"
+               exit -1
+               ;;
+      esac
+done
+
+function install_shaded_library() {
+    if [ -z "$M2_HOME" ]; then
+        echo "Please set M2_HOME"
+        exit 1
+    fi
+    SHADED_JAR="maven-lifecycle-extension-${VERSION}-shaded.jar"
+    echo "Installing ${SHADED_JAR} into ${M2_HOME}/lib/ext"
+    wget http://central.maven.org/maven2/org/arquillian/smart/testing/maven-lifecycle-extension/${VERSION}/${SHADED_JAR}
+    read -r -p "We want to move shaded jar to M2_HOME with sudo. Can we? [y/N] " response
+    case "$response" in
+        [yY][eE][sS]|[yY])
+              sudo mv $SHADED_JAR $M2_HOME/lib/ext
+            ;;
+        *)
+            ;;
+    esac
+}
+
+function install_extension() {
+
+    # Needs to be without new lines, otherwise my sed skills below will fail badly :\
+    EXTENSION="<extension><groupId>org.arquillian.smart.testing</groupId><artifactId>maven-lifecycle-extension</artifactId><version>${VERSION}</version></extension>"
+
+    if [ ! -f .mvn/extensions.xml ]; then
+        echo "<?xml version=\"1.0\" encoding=\"UTF-8\"?>
+            <extensions>
+              ${EXTENSION}
+            </extensions>" > .mvn/extensions.xml
+    else
+        EXTENSION_REGISTERED=$(cat .mvn/extensions.xml | grep 'org.arquillian.smart.testing' -A 2 |  grep -oPm1 "(?<=<version>)[^<]+");
+
+        if [ ! ${EXTENSION_REGISTERED} ]; then
+            EXTENSION=$(echo ${EXTENSION} | sed -e "s#/#\\\/#g");
+            sed -i -E 's/(.*<extensions>)(.*)/\1\n'$EXTENSION'\2/g' .mvn/extensions.xml
+            echo "Installed Smart Testing Extension ${VERSION}"
+        else
+            echo -e "Smart Testing Extension already registered with version ${EXTENSION_REGISTERED}\c"
+
+            if [ ${INSTALL_SPECIFIC_VERSION} -eq 1 ]; then
+                if [ "${EXTENSION_REGISTERED}" != "${VERSION}" ]; then
+                    echo -e " - overwritting with ${VERSION}."
+                    override_version $VERSION
+                    echo -e "Updated Smart Testing Extension to ${VERSION}\c"
+                fi
+                echo "."
+            elif [ $EXTENSION_REGISTERED != $LATEST ]; then
+                read -r -p ". Do you want to override with latest ${LATEST}? [y/N] " response
+                case "$response" in
+                    [yY][eE][sS]|[yY])
+                          override_version ${LATEST}
+                          echo "Updated Smart Testing Extension to ${LATEST}"
+                        ;;
+                    *)
+
+                        ;;
+                esac
+            else
+                echo " which is the latest stable version."
+            fi
+        fi
+    fi
+
+    mv .mvn/extensions.xml .mvn/extensions-unformatted.xml
+    xmllint --format .mvn/extensions-unformatted.xml > .mvn/extensions.xml
```

Comment:
Seems `xmllint` is installed in MacOs and Linux but is this true in all cases?

Reply:
Right, I forgot to clarify it in the docs.

- [ ] Mention `xmllint` as a prerequisite in the docs